### ISSUE TITLE
Unable to list listeners in multiple node cluster.

### ISSUE THREAD

**Issue by ruifung** [emqx/emqx#9733]
### What happened?

When run as a multi node cluster (2 node / 3 node in my case)

The API for listing listeners errors out. This does not occur when only a single node is running.

This causes a few problems, like the emqx operator not generating the service because it depends on that API. Or the dashboard listener configuration page being broken.

### What did you expect to happen?

For it to list the listeners which is *identical* on the entire cluster based on the same emqx manifest.

### How can we reproduce it (as minimally and precisely as possible)?

Run a multiple node emqx cluster using the emqx operator, watch the error occur.

### Anything else we need to know?

```
500 INTERNAL_ERROR:error, badarith, [{erlang,'+',[infinity,infinity],[{error_info,#{module => erl_erts_errors}}]},{emqx_mgmt_api_listeners,max_conn,2,[{file,"emqx_mgmt_api_listeners.erl"},{line,633}]},{emqx_mgmt_api_listeners,format_status,4,[{file,"emqx_mgmt_api_listeners.erl"},{line,623}]},{lists,foldl,3,[{file,"lists.erl"},{line,1267}]},{emqx_mgmt_api_listeners,listener_status_by_id,2,[{file,"emqx_mgmt_api_listeners.erl"},{line,517}]},{emqx_mgmt_api_listeners,listener_status_by_id,1,[{file,"emqx_mgmt_api_listeners.erl"},{line,492}]},{emqx_mgmt_api_listeners,list_listeners,2,[{file,"emqx_mgmt_api_listeners.erl"},{line,364}]},{minirest_handler,apply_callback,3,[{file,"minirest_handler.erl"},{line,111}]},{minirest_handler,handle,2,[{file,"minirest_handler.erl"},{line,44}]},{minirest_handler,init,2,[{file,"minirest_handler.erl"},{line,27}]},{cowboy_handler,execute,2,[{file,"cowboy_handler.erl"},{line,41}]},{cowboy_stream_h,execute,3,[{file,"cowboy_stream_h.erl"},{line,318}]},{cowboy_stream_h,request_process,3,[{file,"cowboy_stream_h.erl"},{line,302}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]
```
and

```
Warning  FailedToGetListenerPorts  2m23s (x7 over 3m45s)  emqx-controller  failed to get API api/v5/listeners, status : 500 Internal ││  Server Error, body: {"code":"INTERNAL_ERROR","message":"error, badarith, [{erlang,'+',[infinity,infinity],[{error_info,#{module => er ││ l_erts_errors}}]},{emqx_mgmt_api_listeners,max_conn,2,[{file,\"emqx_mgmt_api_listeners.erl\"},{line,633}]},{emqx_mgmt_api_listeners,fo ││ rmat_status,4,[{file,\"emqx_mgmt_api_listeners.erl\"},{line,623}]},{lists,foldl,3,[{file,\"lists.erl\"},{line,1267}]},{emqx_mgmt_api_l ││ isteners,listener_status_by_id,2,[{file,\"emqx_mgmt_api_listeners.erl\"},{line,517}]},{emqx_mgmt_api_listeners,listener_status_by_id,1 ││ ,[{file,\"emqx_mgmt_api_listeners.erl\"},{line,492}]},{emqx_mgmt_api_listeners,list_listeners,2,[{file,\"emqx_mgmt_api_listeners.erl\" ││ },{line,364}]},{minirest_handler,apply_callback,3,[{file,\"minirest_handler.erl\"},{line,111}]},{minirest_handler,handle,2,[{file,\"mi ││ nirest_handler.erl\"},{line,44}]},{minirest_handler,init,2,[{file,\"minirest_handler.erl\"},{line,27}]},{cowboy_handler,execute,2,[{fi ││ le,\"cowboy_handler.erl\"},{line,41}]},{cowboy_stream_h,execute,3,[{file,\"cowboy_stream_h.erl\"},{line,318}]},{cowboy_stream_h,reques ││ t_process,3,[{file,\"cowboy_stream_h.erl\"},{line,302}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,226}]}]"}
```

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.0.9
datetime  : 2023-01-12T05:03:44.104335596+00:00
uptime    : 2 minutes, 58 seconds
```

</details>

### OS version

<details>

Not possible to provide because Talos Linux does not provide a shell at all.
Talos Linux 1.3.0
```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>

</details>

**Comment by zhongwencool** [comment]
Hi, Thanks for reporting. we find the change `max_connections = infinity` cause this bug. we plan to fix it on v5.0.15.

Until that is fixed, you can try to set `max_connections=1024000`

**Comment by zhongwencool** [comment]
@ruifung Could you provide the full `bootstrapConfig` in the operator?

**Comment by ruifung** [comment]
@zhongwencool Apologies if this is slightly annoying to read, I pulled it out of the configmap.

```json
{node:{data_dir:data, etc_dir:etc}, cluster:{discovery_strategy:dns, dns:{record_type:srv, name:"emqx-headless.home-network.svc.cluste ││ r.local"}}, dashboard:{listeners:{http:{bind:18083}}, default_username:admin, default_password:public}, listeners:{ssl:{default:{bind: ││ "0.0.0.0:8883", ssl_options:{certfile:"/certs/tls.crt", keyfile:"/certs/tls.key"}}}, quic:{default:{certfile:"/certs/tls.crt", keyfile ││ :"/certs/tls.key", bind:"0.0.0.0:14567"}}, tcp:{default:{bind:"0.0.0.0:1883", max_connections:1024000}}, ws:{default:{bind:"0.0.0.0:80 ││ 83"}}, wss:{default:{bind:"0.0.0.0:8084", ssl_options:{certfile:"/certs/tls.crt", keyfile:"/certs/tls.key"}}}}} 
```

**Comment by zhongwencool** [comment]
@ruifung Before we fix this, please add max_connections=1024000 to every listeners to avoid this issue. such as:
```
{node:{data_dir:data, etc_dir:etc}, cluster:{discovery_strategy:dns, dns:{record_type:srv, name:"emqx-headless.home-network.svc.cluste ││ r.local"}}, dashboard:{listeners:{http:{bind:18083}}, default_username:admin, default_password:public}, listeners:{ssl:{default:{bind: ││ "0.0.0.0:8883", max_connections:102400, ssl_options:{certfile:"/certs/tls.crt", keyfile:"/certs/tls.key"}}}, quic:{default:{certfile:"/certs/tls.crt", keyfile ││ :"/certs/tls.key", bind:"0.0.0.0:14567",max_connections:1024000}}, tcp:{default:{bind:"0.0.0.0:1883", max_connections:1024000}}, ws:{default:{bind:"0.0.0.0:80 ││ 83",max_connections:1024000}}, wss:{default:{bind:"0.0.0.0:8084",max_connections:1024000, ssl_options:{certfile:"/certs/tls.crt", keyfile:"/certs/tls.key"}}}}}
```
